kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/message/MessageEventManager.kt

```diff
@@ -0,0 +1,267 @@
+package com.kotlindiscord.kord.extensions.message
+
+import com.kotlindiscord.kord.extensions.utils.events
+import com.kotlindiscord.kord.extensions.utils.waitFor
+import dev.kord.core.Kord
+import dev.kord.core.behavior.MessageBehavior
+import dev.kord.core.entity.Message
+import dev.kord.core.entity.ReactionEmoji
+import dev.kord.core.event.Event
+import dev.kord.core.event.message.*
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Manager of listeners about the events for a message.
+ * @property message Message responsible of events.
+ * @property timeout Time to stop listening to events after the last received event corresponding to the message.
+ */
+public open class MessageEventManager(
+    public val message: MessageBehavior,
+    private val timeout: Long? = DEFAULT_TIME_LISTENING
+) {
+
+    public companion object {
+        /**
+         * Default time before stop the listening of events.
+         */
+        public const val DEFAULT_TIME_LISTENING: Long = 1000L * 60L * 5L
+    }
+
+    /**
+     * Kord instance.
+     */
+    public val kord: Kord
+        get() = message.kord
+
+    /**
+     * Job running to listen to events.
+     */
+    public var job: Job? = null
+        private set
+
+    /**
+     * `true` if the manager listening the events, `false` otherwise.
+     */
+    @Volatile
```

Comment:
The "Volatile" annotation is not for java users, in fact it is an annotation taken into account by Kotlin for JS, JVM, etc. (cf. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-volatile/). 
I use it because the variable can be both modified or watched by 2 different threads. 
After that it's just to be sure that the information is correctly obtained and updated, it can be a bit overkill